src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs

```diff
@@ -43,7 +45,54 @@ public JniTypeSignature GetTypeSignature (Type type)
 			{
 				AssertValid ();
 
-				return GetTypeSignatures (type).FirstOrDefault ();
+				if (type == null)
```

Comment:
Given the introduced code duplication, we should have a comment (and `#region`?) noting that this block needs to be kept semantically in sync with `GetTypeSignatures()`, so that they don't return different values.